samza-core/src/main/java/org/apache/samza/clustermanager/StandbyContainerManager.java

```diff
@@ -375,24 +454,39 @@ boolean checkStandbyConstraints(String containerIdToStart, String host) {
       SamzaResource resource = samzaApplicationState.pendingProcessors.get(containerID);
 
       // return false if a conflicting container is pending for launch on the host
-      if (resource != null && resource.getHost().equals(host)) {
-        log.info("Container {} cannot be started on host {} because container {} is already scheduled on this host",
-            containerIdToStart, host, containerID);
+      if (!checkStandbyConstraintsHelper(containerIdToStart, host, resource, containerID, "pending")) {
         return false;
       }
 
       // return false if a conflicting container is running on the host
       resource = samzaApplicationState.runningProcessors.get(containerID);
-      if (resource != null && resource.getHost().equals(host)) {
-        log.info("Container {} cannot be started on host {} because container {} is already running on this host",
-            containerIdToStart, host, containerID);
+      if (!checkStandbyConstraintsHelper(containerIdToStart, host, resource, containerID, "running")) {
```

Comment:
I still feel this is unclean to me especially passing in parameters that are not necessarily relevant to the core of what the method is responsible for. e.g., "pending" vs "running" which is purely for logging and isn't state per say and one realizes until reading through the code underneath.

If you don't want to inline and still feel refactor would help and make it clear, i'd suggest extracting the existing logic into `checkActiveAndStandbyOnSameHost` and then add your logic into `checkActiveAndStandbyOnSameFaultDomain` and within `checkStandbyConstraints` you can fire off both of these checks or one of them based on the fault domain enabled or not.

Reply:
Got it. I've inlined the method as you suggested.